packages/yourloops/lib/providers/patient-list.provider.hook.tsx

```diff
@@ -0,0 +1,106 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { useState } from 'react'
+import { type PatientsFilters } from './models/patients-filters.model'
+import { type PatientListContextResult } from './models/patient-list-context-result.model'
+import { PatientListColumns } from '../../components/patient-list/enums/patient-list.enum'
+import { type GridColumnVisibilityModel, useGridApiRef } from '@mui/x-data-grid'
+import { useAuth } from '../auth'
+
+const DEFAULT_FILTERS = {
+  pendingEnabled: false,
+  manualFlagEnabled: false,
+  telemonitoredEnabled: false,
+  timeOutOfTargetEnabled: false,
+  hypoglycemiaEnabled: false,
+  dataNotTransferredEnabled: false,
+  messagesEnabled: false
+}
+
+export const usePatientListProviderHook = (): PatientListContextResult => {
+  const { user, updatePreferences } = useAuth()
+  const gridApiRef = useGridApiRef()
+
+  const getColumnPreference = (ColumnName: PatientListColumns): boolean => {
+    const userPreferredColumns = user.preferences?.patientsListSortedOptionalColumns
+    return userPreferredColumns ? userPreferredColumns.includes(ColumnName) : true
+  }
+
+  const [filters, setFilters] = useState<PatientsFilters>(DEFAULT_FILTERS)
+  const [displayedColumns, setDisplayedColumns] = useState<GridColumnVisibilityModel>({
+    [PatientListColumns.Flag]: true,
+    [PatientListColumns.System]: getColumnPreference(PatientListColumns.System),
+    [PatientListColumns.Patient]: true,
+    [PatientListColumns.TimeOutOfRange]: user.isUserHcp() ? getColumnPreference(PatientListColumns.TimeOutOfRange) : false,
+    [PatientListColumns.SevereHypoglycemia]: user.isUserHcp() ? getColumnPreference(PatientListColumns.SevereHypoglycemia) : false,
+    [PatientListColumns.DataNotTransferred]: user.isUserHcp() ? getColumnPreference(PatientListColumns.DataNotTransferred) : false,
+    [PatientListColumns.LastDataUpdate]: getColumnPreference(PatientListColumns.LastDataUpdate),
+    [PatientListColumns.Messages]: user.isUserHcp() ? getColumnPreference(PatientListColumns.Messages) : false,
+    [PatientListColumns.Actions]: true
+  })
+
+  const buildColumnsPreferencesArray = (columnsVisibilityModel: GridColumnVisibilityModel): string[] => {
+    const columnsPreferences = []
+    for (const columnName in columnsVisibilityModel) {
+      if (columnsVisibilityModel[columnName]) {
+        columnsPreferences.push(columnName)
+      }
+    }
+    return columnsPreferences
+  }
+
+  const saveColumnsPreferences = (updatedColumnsVisibilityModel: GridColumnVisibilityModel): void => {
+    setDisplayedColumns(updatedColumnsVisibilityModel)
+    updatePreferences({ patientsListSortedOptionalColumns: buildColumnsPreferencesArray(updatedColumnsVisibilityModel) })
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (blocking):** ‏Forgot the await there